### PR TITLE
average.go

### PR DESCRIPTION
**Pull request by egor-geller**

### CODE REVIEW

File: average.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import "fmt"
+
+func average(x []int) float64 {
+	var sum, N int
+	sum = 0
+	N = len(x)
+	for i := 0; i < N; i++ {
+		sum += x[i]
+	}
+	return float64(sum) / float64(N)
```
Comment (imatsiash-ib) [inline average.go]: What happens if N == 0 ?
Please add a check that slice length is greater than 0.

File: average.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+func average(x []int) float64 {
+	var sum, N int
+	sum = 0
```
Comment (imatsiash-ib) [inline average.go]: The default value of `sum` is 0.
This line is redundant.

File: average.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	N = len(x)
+	if N <= 0 {
+		fmt.Print("Slice has no elements")
+		os.Exit(1)
```
Comment (imatsiash-ib) [inline average.go]: Will be better to return an error in this case. Or just return 0.

**Review by imatsiash-ib — APPROVED** [review]

**Review by burov — APPROVED** [review]
Looks ok for me

**Pull request closed without merging**